pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -634,154 +635,159 @@ def getData(self):
+        x = self.xData
+        y = self.yData
+        view = self.getViewBox()
+        if view is None:
+            v_rng = None
+        else:
+            v_rng = self.getViewBox().viewRect() # this is always up-to-date
+        if v_rng is None:
+            v_rng = self.viewRect()
+            # assert v_rng is None # halt if this actually did anything!
+
+        if self.opts['fftMode']:
+            x,y = self._fourierTransform(x, y)
+            # Ignore the first bin for fft data if we have a logx scale
+            if self.opts['logMode'][0]:
+                x=x[1:]
+                y=y[1:]
+
+        if self.opts['derivativeMode']:  # plot dV/dt
+            y = np.diff(self.yData)/np.diff(self.xData)
+            x = x[:-1]
+        if self.opts['phasemapMode']:  # plot dV/dt vs V
+            x = self.yData[:-1]
+            y = np.diff(self.yData)/np.diff(self.xData)
+                
+        with np.errstate(divide='ignore'):
+            if self.opts['logMode'][0]:
+                x = np.log10(x)
+            if self.opts['logMode'][1]:
+                if np.issubdtype(y.dtype, np.floating):
+                    eps = np.finfo(y.dtype).eps
+                else:
+                    eps = 1
+                y = np.sign(y) * np.log10(np.abs(y)+eps)
+
+        ds = self.opts['downsample']
+        if not isinstance(ds, int):
+            ds = 1
+
+        if self.opts['autoDownsample']:
+            # this option presumes that x-values have uniform spacing
+            if v_rng is not None and len(x) > 1:
+                dx = float(x[-1]-x[0]) / (len(x)-1)
+                if dx != 0.0:
+                    x0 = (v_rng.left()-x[0]) / dx
+                    x1 = (v_rng.right()-x[0]) / dx
+                    width = self.getViewBox().width()
+                    if width != 0.0:
+                        ds = int(max(1, int((x1-x0) / (width*self.opts['autoDownsampleFactor']))))
+                    ## downsampling is expensive; delay until after clipping.
+
+        if self.opts['clipToView']:
+            if view is None or view.autoRangeEnabled()[0]:
+                pass # no ViewBox to clip to, or view will autoscale to data range.
+            else:
+                # clip-to-view always presumes that x-values are in increasing order
+                if v_rng is not None and len(x) > 1:
+                    # print('search:', v_rng.left(),'-',v_rng.right() )
+                    # find first in-view value (left edge) and first out-of-view value (right edge)
+                    # since we want the curve to go to the edge of the screen, we need to preserve
+                    # one down-sampled point on the left and one of the right, so we extend the interval
+                    x0 = np.searchsorted(x, v_rng.left()) - ds
+                    x0 = fn.clip_scalar(x0, 0, len(x)) # workaround
+                    # x0 = np.clip(x0, 0, len(x))
+
+                    x1 = np.searchsorted(x, v_rng.right()) + ds
```

Comment:
This was previously adding 2*ds, but I think one down-sampling block is sufficient here.